test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,174 +130,122 @@ public ServerSocket createServerSocket(int port) throws IOException {
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
+    public void testRmiCommunication(RMIServerSocketFactory serverFactory, boolean expectException) {
+
+        HelloImpl server = null;
+        try {
+            server = new HelloImpl(0,
+                                    new ClientFactory(),
+                                    serverFactory);
+            Remote stub = server.runServer();
+            HelloClient client = new HelloClient();
+            client.runClient(stub);
+            if (expectException) {
+                throw new RuntimeException("Test completed without throwing an expected exception.");
             }
-            break;
-        case 3: /* needClientAuth=true */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            null,
-                                            null,
-                                            true));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg2);
-            } catch (Exception e) {
-                System.out.println(msg3 + " Exception: " + e.toString());
-                e.printStackTrace(System.out);
-                System.exit(1);
-            }
-            break;
-        case 4: /* server side dummy_ciphersuite */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(SSLContext.getDefault(),
-                                            new String[] {"dummy_ciphersuite"},
-                                            null,
-                                            false));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+
+        } catch (IOException | IllegalArgumentException exc) {
+            if (!expectException) {
+                throw new RuntimeException("An error occurred during test execution", exc);
+            } else {
+                System.out.println("Caught expected exception: " + exc);
```

Comment:
I'm not quite clear on what the expected exception actually is -- is it an IllegalArgumentException or an IOException? Or does it differ from case to case? It would be good to be precise about exactly which exception is expected to be thrown from which test case. If IOException is caught here only because it's declared to be thrown by some method, and it's incidental to the test, I'd suggest it be propagated. (See my similar comments on testServerFactory.)